mmengine/config/utils.py

```diff
@@ -144,3 +149,301 @@ def visit_Assign(self, node):
+                >>> global_dict.update(dataset=base_dict['.._base_.datasets.coco_detection']['dataset'])  # only update `dataset`
+    """  # noqa: E501
+
+    def __init__(self,
+                 global_dict: dict,
+                 base_dict: Optional[dict] = None,
+                 filename: Optional[str] = None):
+        self.base_dict = base_dict if base_dict is not None else {}
+        self.global_dict = global_dict
+        # In Windows, the filename could be like this:
+        # "C:\\Users\\runneradmin\\AppData\\Local\\"
+        # Although it has been an raw string, ast.parse will firstly escape
+        # it as the executed code:
+        # "C:\Users\runneradmin\AppData\Local\\\"
+        # As you see, the `\U` will be treated as a part of
+        # the escape sequence during code parsing, leading to an
+        # parsing error
+        # Here we use `encode('unicode_escape').decode()` for double escaping
+        if isinstance(filename, str):
+            filename = filename.encode('unicode_escape').decode()
+        self.filename = filename
+        self.imported_obj: set = set()
+        super().__init__()
+
+    def visit_ImportFrom(
+        self, node: ast.ImportFrom
+    ) -> Optional[Union[List[ast.Assign], ast.ImportFrom]]:
+        """Hack the ``from ... import ...`` syntax and update the global_dict.
+
+        Examples:
+            >>> from mmdet.models import RetinaNet
+
+        Will be parsed as:
+
+        Examples:
+            >>> RetinaNet = lazyObject('mmdet.models', 'RetinaNet')
+
+        ``global_dict`` will also be updated by ``base_dict`` as the
+        class docstring says.
+
+        Args:
+            node (ast.AST): The node of the current import statement.
+
+        Returns:
+            Optional[List[ast.Assign]]: There three cases:
+
+                * If the node is a statement of importing base files.
+                  None will be returned.
+                * If the node is a statement of importing a builtin module,
+                  node will be directly returned
+                * Otherwise, it will return the assignment statements of
+                  ``LazyObject``.
+        """
+        # Built-in modules will not be parsed as LazyObject
+        module = f'{node.level*"."}{node.module}'
+        if _is_builtin_module(module):
+            return node
+
+        if module in self.base_dict:
+            for alias_node in node.names:
+                if alias_node.name == '*':
+                    self.global_dict.update(self.base_dict[module])
+                    return None
+                if alias_node.asname is not None:
+                    base_key = alias_node.asname
+                else:
+                    base_key = alias_node.name
+                self.global_dict[base_key] = self.base_dict[module][
+                    alias_node.name]
+            return None
+
+        # TODO: Support lazyimport module from relative path
```

Comment:
```suggestion
        # TODO: Support lazily importing modules from a relative path
```

Reply:
Actually, this have been implemented, just remove it.